### ISSUE TITLE
ES6 & Beyond, Chapter 3: default export of a function expression

### ISSUE THREAD

**Issue by dongbohu** [getify/You-Dont-Know-JS#1186]
Lines 1280-1286 of `ch3.md` in `ES6 & Beyond`:
```
By the way, the first snippet could also have been written as:

export default function foo(..) {  
	// ..
}
```
(From now on let's call this snippet **revised snippet**.)

And here is the first snippet on the book:
```js
function foo(..) {  
	// ..
}
export default foo; 
```
These two default exports are NOT exactly equivalent. Instead, the first snippet should be equivalent to:
```js
export default (function foo(..) {  
	// ..
})
```
(Note the parenthesis around the function expression.) Both default exports are static bindings to a function expression.

And the **revised snippet** is actually equivalent to the second snippet on the book:
```js
function foo(..) {
	// ..
}
export { foo as default };
```
Both default exports are dynamic bindings to `foo`.

**Comment by kruchy8** [comment]
I agree, but with your equivalent proposition we don't actually get an identifier that is accessible in a module's scope, so we cannot reassign it (that's what this paragraph is about). I guess the only way to export a binding to a function expression using it's identifier is `export default foo;` ?

**Comment by getify** [comment]
> These two default exports are NOT exactly equivalent

I'm not sure I agree. At least as far as I can tell, ES6 modules made exceptions for the `export` / `export default` statements, that when used with something that otherwise would look like a function declaration (not an function expression), it would still be treated that way.

In fact, they also had to create the perplexing "anonymous function declaration" cover for that form.

IOW, I don't think these are strictly treated as function expressions, I think they are indeed function declarations. The main observational point there would be, in both forms, is the lexical name `foo` hoisted and available throughout the scope. I believe it is.
